aiokafka/conn.py

```diff
@@ -227,11 +227,15 @@ async def connect(self):
             self._idle_handle = loop.call_soon(
                 self._idle_check, weakref.ref(self))
 
-        if self._version_hint and self._version_hint >= (0, 10):
-            await self._do_version_lookup()
+        try:
+            if self._version_hint and self._version_hint >= (0, 10):
+                await self._do_version_lookup()
 
-        if self._security_protocol in ["SASL_SSL", "SASL_PLAINTEXT"]:
-            await self._do_sasl_handshake()
+            if self._security_protocol in ["SASL_SSL", "SASL_PLAINTEXT"]:
+                await self._do_sasl_handshake()
+        except:  # noqa: E722
```

Comment:
Please change this to ``except Exception``, I would prefer not to have an implicit catch on BaseException if there is no need for it.

Reply:
We do need to catch BaseException. Actually I stepped into the problem with CancelledError, which nowadays doesn't inherit from Exception.